src/fu-util.c

```diff
@@ -2067,6 +2070,18 @@ fu_util_update_all (FuUtilPrivate *priv, GError **error)
 			continue;
 		}
 		rel = g_ptr_array_index (rels, 0);
+
+		uri_tmp = fwupd_release_get_uri (rel);
+		name_tmp = g_hash_table_lookup (uris, uri_tmp);
+		if (name_tmp != NULL) {
+			g_debug ("skipping %s due to %s already in list for %s",
+				fu_device_get_name (dev),
+				uri_tmp, name_tmp);
+			continue;
+		}
+		g_hash_table_insert (uris,
+				     (gpointer) uri_tmp,
+				     (gpointer) fu_device_get_name (dev));
```

Comment:
What's this chunk doing? It's somewhat inelegant.

Reply:
1. Fetching URI for each device that has an update available
2. If the URI is the same for two devices, skipping "updating that device separately"

So if you have
Device X: 1.01
Device Y: 1.26

CAB Z with:
Release for X: 1.02
Release for Y: 1.27

As you walk through the devices with updates you'll see that X and Y both point to that same URI.  So X will perform the update using the CAB on all supported devices from the CAB.  Then it will come to Y and since it was the same URI skip it.